File: Advent of Code 2022/day1-p2.exs
defmodule AVS do
    def main do
        problem_file = "input/day1.txt"

        lines = File.read!(problem_file) |> String.split("\n")
        food = sum_by_group(lines)
        result = Enum.sort(food) |> Enum.take(-3) |> Enum.sum()
        IO.inspect(result)
    end

    def sum_by_group(lines, sum \\ 0)

    def sum_by_group([], _), do: []

    def sum_by_group([line | lines], sum) do
        if String.trim(line) === "" do
            [sum | sum_by_group(lines)]
        else
            sum_by_group(lines, sum + String.to_integer(line))
        end
    end
end

AVS.main()

File: Advent of Code 2021/day1-p2.py
problem_file_path = "input/d1.txt"

with open(problem_file_path, "r") as problem_file:
	values = [int(i) for i in problem_file]


count = -1
last_window = -1

for i in range(len(values) - 2):
	window = sum(values[i:i + 3])
	if window > last_window:
		count += 1
	last_window = window

print(count)
